client/my-sites/email/inbox/mailbox-selection-list/index.jsx

```diff
@@ -131,14 +131,16 @@ MailboxItem.propType = {
 	key: PropTypes.string,
 };
 
-const NewMailboxUpsell = () => {
+const NewMailboxUpsell = ( { domain } ) => {
 	const translate = useTranslate();
 	const selectedSite = useSelector( getSelectedSite );
 	const selectedSiteSlug = selectedSite?.slug;
 
 	const handleCreateNewMailboxClick = useCallback( () => {
 		recordInboxNewMailboxUpsellClickEvent();
-		page( emailManagement( selectedSiteSlug, null, null, { source: INBOX_SOURCE } ) );
+		page(
```

Comment:
I think it would be nice to take this opportunity to pass this path as an `href` prop to `Button` instead of using the `page` function to navigate programmatically.

We could still use the `onClick` handler for tracking the click event.

Reply:
I left `page` in place. I don't actually know what `page` does 😆 but I wonder if it applies some sort of delay to ensure our tracking event gets sent before navigating away from the current page? Is `page` a third party library?

Reply:
I followed up in a new thread!